AUTHORS.txt

```diff
@@ -0,0 +1,25 @@
+The Spyder Project Contributors is composed of:
+
+* Pierre Raybaut <pierre.raybaut@gmail.com> (Main Spyder author)
+* Carlos Cordoba <ccordoba12@gmail.com> (Current maintainer)
+* All other developers that have contributed to the spyder repository:
+  <https://github.com/spyder-ide/spyder/graphs/contributors>
+
+Additionally, some assets and code were originally sourced from third-party
+authors or projects, including:
```

Comment:
These are not *authors* of the project, at least in the way we mean here. So I think all this text should be removed. 

Developers of the several projects we have taken code from (like QCrash or gtabview) are more *authors* of the Spyder source code than Icon developers (I know that's debatable, but this addition doesn't make any sense to me).

Reply:
> Developers of the several projects we have taken code from (like QCrash or gtabview) are more authors of the Spyder source code than Icon developers

Indeed, I would normally agree. However, there is a specific reason this is here—the licenses of at least half (2/4) the external icon sources (CC-BY) require an attribution credit wherever other authors of the covered work are normally listed, and additionally both authors if I recall explicitly requested it (that's also why I added it them to the README and kept them in the About screen, where they were already listed); the LGPL (which the other half of the icons are under) also contains a somewhat less clear clause that implies something similar. Even if some didn't require it (IANAL, after all), it would be equally inconsistent to only list certain icon authors and not others so it made sense to list all of them.

However, if you have a better idea that would still meet the legal and ethical requirements, I'm certainly open to hearing it.

Reply:
> gtabview

I didn't find anything under that name in my search of ``3.x`` for third-party material, and searching for that right now reveals no hits, unlike in master where I can see it is clearly marked. Is the code in ``3.x`` also and someone was delinquent about adding the appropriate copyright notices, disclaimer and license to that branch, or is it only in ``master`` and I'll need to do it after this is merged there?

Reply:
Perhaps it is just the word "authors" that is the issue here, and we should call them "designers" or "artists".

Generally, I think we should be generous in giving credit.

Reply:
Correct me if I'm wrong, but if I understand @ccordoba12 's point correctly, its really that we mention the icon authors there, but not those of the external source code we've used, which may arguably be seen as even more central to Spyder. I agree that its inconsistent, but including the icon authors is motivated by the requirements of the licenses and their specific requests (and they are also the ones listed in the About dialog already), while if we were to include all the authors of the code as well, it would be much longer and rather duplicative of the NOTICE.txt file. However, on the other hand, it could be seen as a convenient summary of the 4000-line latter, so if that's what's agreed on then I can do that.

Reply:
> I agree that its inconsistent, but including the icon authors is motivated by the requirements of the licenses and their specific requests

Ok, ugly stuff, but let's leave it to comply with the requirements.

Reply:
> Generally, I think we should be generous in giving credit.

Sure, but calling *authors* to people who has not put a single line of code here it's just really misleading.

Reply:
> Sure, but calling authors to people who has not put a single line of code here it's just really misleading.

Well, perhaps its different in different languages, but legally- and semantically-speaking the creator of an icon is its "author" as much as a developer writing a line of code is its "author" (in some jurisdictions, even more so); and both contribute to the final product presented to the user, the former typically more visibly so.

Reply:
I don't want to keep arguing about this.

Reply:
You're right, sorry.